src/main/java/cloud/eppo/EppoClient.java

```diff
@@ -0,0 +1,477 @@
+  private static final boolean DEFAULT_IS_GRACEFUL_MODE = true;
+
+  private final ConfigurationRequestor requestor;
+  private final AssignmentLogger assignmentLogger;
+  private final boolean isGracefulMode;
+  private final String sdkName;
+  private final String sdkVersion;
+  private final boolean isConfigObfuscated;
+
+  private static EppoClient instance;
+
+  // Fields useful for testing in situations where we want to mock the http client or configuration
+  // store (accessed via reflection)
+  /**
+   * @noinspection FieldMayBeFinal
+   */
+  private static EppoHttpClient httpClientOverride = null;
+
+  /**
+   * @noinspection FieldMayBeFinal
+   */
+  private static ConfigurationStore configurationStoreOverride = null;
+
+  private EppoClient(
+      String apiKey,
+      String sdkName,
+      String sdkVersion,
+      String host,
+      AssignmentLogger assignmentLogger,
+      boolean isGracefulMode
+  ) {
+    ConfigurationStore configStore =
+        configurationStoreOverride == null
+            ? new ConfigurationStore()
+            : configurationStoreOverride;
+
+    EppoHttpClient httpClient = buildHttpClient(host, apiKey, sdkName, sdkVersion);
+    requestor = new ConfigurationRequestor(configStore, httpClient);
+    this.assignmentLogger = assignmentLogger;
+    this.isGracefulMode = isGracefulMode;
+    // Save SDK name and version to include in logger metadata
+    this.sdkName = sdkName;
+    this.sdkVersion = sdkVersion;
+    // For now, the configuration is only obfuscated for Android clients
+    this.isConfigObfuscated = sdkName.toLowerCase().contains("android");
+    // TODO: caching initialization (such as setting an API-key-specific prefix
+  }
+
+  private EppoHttpClient buildHttpClient(String host, String apiKey, String sdkName, String sdkVersion) {
+    EppoHttpClient httpClient;
+    if (httpClientOverride != null) {
+      // Test/Debug - Client is mocked entirely
+      httpClient = httpClientOverride;
+    } else {
+      // Normal operation
+      httpClient = new EppoHttpClient(host, apiKey, sdkName, sdkVersion);
+    }
+    return httpClient;
+  }
+
+  public static EppoClient init(
+      String apiKey,
+      String sdkName,
+      String sdkVersion,
+      String host,
+      AssignmentLogger assignmentLogger,
+      boolean isGracefulMode
+  ) {
+
+    if (apiKey == null) {
+      throw new IllegalArgumentException("Unable to initialize Eppo SDK due to missing API key");
+    }
+    if (sdkName == null || sdkVersion == null) {
+      throw new IllegalArgumentException("Unable to initialize Eppo SDK due to missing SDK name or version");
+    }
+
+    if (instance != null) {
+      // TODO: also check we're not running a test
+      log.warn("Reinitializing an Eppo Client instance that was already initialized");
+    }
+    instance = new EppoClient(apiKey, sdkName, sdkVersion, host, assignmentLogger, isGracefulMode);
+    instance.refreshConfiguration();
+
+    return instance;
+  }
+
+  /**
```

Comment:
The rest of this file is mostly unchanged from the Android implementation ([source](https://github.com/Eppo-exp/android-sdk/blob/main/eppo/src/main/java/cloud/eppo/android/EppoClient.java#L132)), with the exception of requiring a default value and using Jackson JSON types instead of build-in/Android's. (Is this something we want to revisit for the common SDK?)

Reply:
I think jackson works well in both scenarios (android and jvm server)